### PR TITLE
chore: run engine tests in parallel 

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15487.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

File: engine/Makefile [hunk 1]
```diff
@@ -150,8 +150,11 @@ ${HOME}/.cds/cdn.tests.cfg.json:
 		@mkdir -p ${HOME}/.cds
 		@echo "{\"dbDriver\": \"postgres\", \"dbUser\" :\"$(TEST_DB_USER)\", \"dbPassword\" :\"$(TEST_DB_PASS)\" ,\"dbName\": \"$(TEST_DB_NAME)\", \"dbSchema\": \"$(TEST_CDN_DB_SCHEMA)\", \"dbHost\": \"$(TEST_DB_HOST)\", \"dbPort\" : \"$(TEST_DB_PORT)\", \"sslMode\": \"disable\", \"redisHost\" : \"$(TEST_REDIS_HOST)\", \"redisPassword\" : \"$(TEST_REDIS_PASSWORD)\"}" > ${HOME}/.cds/cdn.tests.cfg.json
 
-test: test-config clean
-	@for PKG in $(TEST_PKGS); do \
+# Build a list of the first level of package in the 'engine' package, then remove the '.', 'dist' and 'sql'
+TEST_COMPONENTS = $(addprefix test-component-,$(filter-out sql,$(filter-out dist,$(filter-out .,$(shell find . -type d -maxdepth 1 -exec basename {} \;)))))
+$(TEST_COMPONENTS):
+	@TEST_PKGS=$$(go list ./... | grep $(subst test-component-,,$@)); \
```
Comment (richardlt) [inline engine/Makefile:156]: Can `cd` in dir to exec go list. This will avoid problems if a child package has the same name as one of parent dir.

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15490.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Review by yesnault — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15492.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Review by sguiheux — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15492.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✘  
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

File: engine/Makefile (hunk 1 above)
Comment (fsamin) [reply engine/Makefile]: done

**Review by richardlt — APPROVED** [review]

**Pull request merged**